Reqnroll/Bindings/BindingDelegateInvoker.cs

```diff
@@ -25,22 +25,31 @@ public virtual async Task<object> InvokeDelegateAsync(Delegate bindingDelegate,
 
             try
```

Comment:
This is a great idea! Basically you have introduced the ExecutionContext capture logic also for the case when the ExecutionContext already exists, because the current solution captures it "too late", it has to be captured **within** the   `ExecutionContext.Run`.

But now, as we have the idea, this can be further simplified, because currently the capture logic is duplicated. What you can do is to undo the changes of `InvokeInExecutionContext` and perform the try-finally logic in the delegate that is passed to that method. So basically we need to move the try-finally one more step inside. So at the end, the `InvokeDelegateAsync` method return statement will look like:

```
            return await InvokeInExecutionContext(executionContext?.Value, () =>
            {
                try
                {
                    return CreateDelegateInvocationTask(bindingDelegate, invokeArgs);
                }
                finally
                {
                    if (executionContext != null)
                        executionContext.Value = ExecutionContext.Capture();
                }
            });
```

I made a quick check locally with your improved unit tests and they pass with this code. Could you please also try?

Reply:
I also tested the suggestions and it looks good to me.
The code is also simpler. Thanks 🙂